rotkehlchen/exchanges/bitfinex.py

```diff
@@ -0,0 +1,933 @@
+        (from raw_result[10]) over the pair (from raw_result[1]).
+
+        Known pairs format: 'ETHUST', 'ETH:UST'.
+
+        Can raise DeserializationError.
+
+        Schema reference in:
+        https://docs.bitfinex.com/reference#rest-auth-trades
+        """
+        amount = deserialize_asset_amount(raw_result[4])
+        trade_type = TradeType.BUY if amount >= ZERO else TradeType.SELL
+        bfx_pair = raw_result[1]
+        bfx_fee_currency_symbol = raw_result[10]
+        if bfx_pair.startswith(bfx_fee_currency_symbol):
+            regex = re.compile(fr'^({bfx_fee_currency_symbol})\W*(\w+)$')
+        elif bfx_pair.endswith(bfx_fee_currency_symbol):
+            regex = re.compile(fr'^(\w+)\W*({bfx_fee_currency_symbol})$')
+        else:
+            raise DeserializationError(
+                f'Could not deserialize bitfinex trade pair {raw_result[1]} '
+                f'using fee_currency symbol {bfx_fee_currency_symbol}. '
+                f'Raw trade: {raw_result}',
+            )
+
+        match = regex.match(bfx_pair)
+        if match is None:
+            raise DeserializationError(
+                f'Unexpected error deserializing bitfinex trade pair: {raw_result[1]} '
+                f'using fee_currency symbol {bfx_fee_currency_symbol}. '
+                f'Trade pair does not match the pattern. Raw trade: {raw_result}',
+            )
+
+        bfx_base_asset_symbol, bfx_quote_asset_symbol = match.groups()
+        try:
+            base_asset = asset_from_bitfinex(
+                bitfinex_name=bfx_base_asset_symbol,
+                currency_map=currency_map,
+            )
+            quote_asset = asset_from_bitfinex(
+                bitfinex_name=bfx_quote_asset_symbol,
+                currency_map=currency_map,
+            )
+        except (UnknownAsset, UnsupportedAsset) as e:
+            asset_tag = 'Unknown' if isinstance(e, UnknownAsset) else 'Unsupported'
+            raise DeserializationError(
+                f'{asset_tag} {e.asset_name} found while processing trade pair due to: {str(e)}',
+            ) from e
+
+        trade = Trade(
+            timestamp=Timestamp(int(raw_result[2] / 1000)),
+            location=Location.BITFINEX,
+            pair=TradePair(f'{base_asset.identifier}_{quote_asset.identifier}'),
+            trade_type=trade_type,
+            amount=AssetAmount(abs(amount)),
+            rate=deserialize_price(raw_result[5]),
+            fee=Fee(abs(deserialize_fee(raw_result[9]))),
+            fee_currency=quote_asset if trade_type == TradeType.BUY else base_asset,
+            link=str(raw_result[0]),
+            notes='',
+        )
+        return trade
+
+    @staticmethod
+    def _get_error_response_data(response_list: List[Any]) -> ErrorResponseData:
+        """Given an error response, return the error code and the reason.
+        """
+        error_code = None
+        reason = None
+        if len(response_list) > 1:
```

Comment:
Is the absence of elif here intentional? I guess probably yes? But I just wanna make sure.

Reply:
yes!